RiotSwiftUI/Modules/Room/LocationSharing/View/LocationSharingUserMarkerView.swift

```diff
@@ -27,19 +27,35 @@ struct LocationSharingUserMarkerView: View {
     
     @State private var frame: CGRect = .zero
     
+    private var usernameColorGenerator: UserNameColorGenerator {
+        let usernameColorGenerator = UserNameColorGenerator()
+        let theme = ThemeService.shared().theme
+        usernameColorGenerator.defaultColor = theme.textPrimaryColor
+        usernameColorGenerator.userNameColors = theme.userNameColors
+        return usernameColorGenerator
+    }
+    
     // MARK: Public
     
+    let isMarker: Bool
```

Comment:
Can we add a small comment to indicate what does it mean

Reply:
I think we can create a reusable `BorderedAvatarImage` that have a `UserNameColorGenerator` property in view model or view. And then integrate this `BorderedAvatarImage` in `LocationSharingUserMarkerView` as subview. `LocationSharingUserMarkerView` and `BorderedAvatarImage` shared the same `UserNameColorGenerator` instance to set the right color.